recipes/_master_package.rb

```diff
@@ -35,6 +35,20 @@
     version node['jenkins']['master']['version']
   end
 
+  bash "Modify packaged log rotation directory" do
+    code <<-EOH
+      sed -i "s|/var/log/jenkins|#{node['jenkins']['master']['log_directory']}|g" /etc/logrotate.d/jenkins
+    EOH
+  end
+
+  bash "Modify packaged log rotation file name" do
+    code <<-EOH
+      if [ -z "`grep #{node['jenkins']['master']['access_log']} /etc/logrotate.d/jenkins`" ]; then
+        sed -i "s|{|#{node['jenkins']['master']['log_directory']}/#{node['jenkins']['master']['access_log']} {|" /etc/logrotate.d/jenkins
+      fi
+    EOH
+  end
```

Comment:
Missing not_if and/or only_if guards to prevent this from executing on every chef-client run